runtime/src/bank/builtins/core_bpf_migration/source_bpf_upgradeable.rs

```diff
@@ -0,0 +1,365 @@
+use {
+    super::error::CoreBpfMigrationError,
+    crate::bank::Bank,
+    solana_sdk::{
+        account::{AccountSharedData, ReadableAccount},
+        bpf_loader_upgradeable::{
+            get_program_data_address, UpgradeableLoaderState, ID as BPF_LOADER_UPGRADEABLE_ID,
+        },
+        pubkey::Pubkey,
+    },
+};
+
+/// Used to validate a source BPF Upgradeable program's account and data
+/// account before migrating a built-in program to Core BPF.
+#[derive(Debug)]
+pub(crate) struct SourceProgramBpfUpgradeable {
```

Comment:
```suggestion
pub(crate) struct SourceUpgradeableBpfProgram {
```

Reply:
Or...
```suggestion
pub(crate) struct SourceUpgradeableBpf {
```

Reply:
I could be talked down here (I know you're emulating `BpfUpgradeableLoader`), but this seems to make more sense to my brain.

Reply:
I also prefer without the suffix, yeah. I'll change both.

Reply:
True I was emulating the loader naming convention, but I don't feel strongly either way. I'm going to go with your more brief suggestion.